controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -252,7 +252,7 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases() (reconcile.Resul
 				r.log.Error(fmt.Errorf("storageClassClaim is not compatible with existing StorageClass"),
 					"StorageClassClaim validation failed.")
 				r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimFailed
-				return reconcile.Result{}, nil
+				return reconcile.Result{}, fmt.Errorf("failed to validate storageclassclaim, %t %s %s", scHasEncryptionMethod, sccEncryptionMethod, scEncryptionMethod)
```

Comment:
This change is wrong it will trigger a reconcile, while no reconcile can help in this case. Please revert.

Reply:
reverted but added the log in above since it will help in debug error from next time